conftest.py

```diff
@@ -88,21 +88,20 @@ def set_env_variables():
 class ArchiverSetup:
     EXE: str = None  # python source based
     FORK_DEFAULT = False
-    prefix = ""
     BORG_EXES = []
 
     def __init__(self):
         self.archiver = None
-        self.tmpdir = str
-        self.repository_path = str
-        self.repository_location = str
-        self.input_path = str
-        self.output_path = str
-        self.keys_path = str
-        self.cache_path = str
-        self.exclude_file_path = str
-        self.patterns_file_path = str
-        self.old_wd = str
+        self.tmpdir: str
+        self.prefix = ""
+        self.repository_path: str
+        self.repository_location: str
+        self.input_path: str
+        self.output_path: str
+        self.keys_path: str
+        self.cache_path: str
+        self.exclude_file_path: str
+        self.patterns_file_path: str
```

Comment:
Hmm, so that declares a type, but does not initialize the attribute? But guess we rather want to also have them initialized.

Reply:
Would it be proper to initialize them to an emptystring `: str = ""`?

I attempted the recommendation of `: str = None` but received the warning "expected type 'str', got 'None' instead"

Reply:
If one does not yet have a valid str value to initialise to, one must use `None`, so that any usage before the real initialisation happened will blow up early with a TypeError (and that would be noticed as a code bug). Initialising to "" would just lead to strange / unclear (and "late") errors.

Reply:
IIRC, type is `Optional[str]` then.

Reply:
> If one does not yet have a valid str value to initialise to, one must use None, so that any usage before the real initialisation happened will blow up early with a TypeError (and that would be noticed as a code bug). Initialising to "" would just lead to strange / unclear (and "late") errors.

That makes a lot of sense, I appreciate the explanation!

>IIRC, type is Optional[str] then.

Ah, yes, I have used this Optional type before and should have thought about that. Thank you!